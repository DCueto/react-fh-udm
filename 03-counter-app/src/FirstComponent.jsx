import PropTypes from 'prop-types';

export const FirstComponent = ( { title, number, subTitle, name } ) => {

  // if( !title ){
  //   console.error('El title no existe');
  // }

  return (
    <div className='card'>
     <h2>{ title }</h2>
     <h3>{ number * 2 }</h3>
     <h4>{ subTitle }</h4>
     <h5>{ name }</h5>
    </div>
  )
}

FirstComponent.propTypes = {
  title: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired,
  subTitle: PropTypes.string
}

FirstComponent.defaultProps = {
  title: 'No hay título',
  subTitle: 'No hay subtítulo',
  name: 'Daniel Cueto'
}