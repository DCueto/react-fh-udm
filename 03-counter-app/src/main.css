html, body{
  background-color: #21232A;
  color: white;
  font-family: Helvetica, Arial, Helvetica, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

.card{
  background-color: #6E8898;
  padding: 20px;
  border-radius: 15px;
  margin: 20px 0;
}

button{
  padding: 15px;
  font-size: 1.3rem;
  margin-right: 5px;
}