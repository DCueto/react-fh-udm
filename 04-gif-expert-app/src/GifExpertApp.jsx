import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

const GifExpertApp = () => {

  const [categories, setCategories] = useState([ 'One Punch' ]);

  const onAddCategory = ( newCategory ) => {
    if( categories.includes( newCategory ) ) return;

    setCategories( [ newCategory, ...categories ] );
  }

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory
        // setCategories={ setCategories }
        onNewCategory = { onAddCategory }
      />

      <section>
        {
          categories.map( (item) =>
            ( <GifGrid
                key={ item }
                category={ item } />
            )
          )
        }
      </section>

    </>
  )

}

export default GifExpertApp;