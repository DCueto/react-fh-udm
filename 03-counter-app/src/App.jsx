import { FirstComponent } from "./FirstComponent";
import { CounterApp } from "./CounterApp";

function App(){
  return (
    <>
      {/* <FirstComponent title='Hola, soy Goku!' number={ 123 } />
      <FirstComponent number={ 123 } /> */}

      <CounterApp value={20} />
    </>
  );
}

export default App;