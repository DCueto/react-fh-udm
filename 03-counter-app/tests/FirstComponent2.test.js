import { render, screen } from '@testing-library/react';
import { FirstComponent } from '../src/FirstComponent';

describe('Pruebas en <FirstComponent />', () => {

  const title = 'Soy un test del título';
  const number = 1;
  const subtitle = 'Soy un subtítulo';

  test('Debe hacer match con el snapshot', () => {

    const { container } = render(
      <FirstComponent title={title} number={number} />
    );

    expect( container ).toMatchSnapshot();

  });

  test('Debe mostrar el mensaje "Soy un test del título"', () => {

    render(
      <FirstComponent title={title} number={number} />
    );
    // screen.debug();
    expect( screen.getByText(title) ).toBeTruthy();

  });

  test('Debe de mostrar el título en un h2', () => {

    render(
      <FirstComponent title={title} number={number} />
    );
    expect( screen.getByRole('heading', {level: 2}).innerHTML )
      .toContain(title);

  });

  test('Debe de mostrar el subtítulo enviado por props', ( ) => {

    render(
      <FirstComponent title={title} number={number} subTitle={subtitle} />
    );
    
    expect( screen.getAllByText(subtitle).length ).toBe(2);
    
  });

});