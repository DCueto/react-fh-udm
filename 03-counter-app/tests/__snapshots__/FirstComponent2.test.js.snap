// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstComponent /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <h2
      data-testid="test-title"
    >
       
      Soy un test del título
       
    </h2>
    <h3>
      2
    </h3>
    <h4>
      No hay subtítulo
    </h4>
    <h4>
      No hay subtítulo
    </h4>
    <h5>
      Daniel Cueto
    </h5>
  </div>
</div>
`;
